src/js/components/AppListItemComponent.jsx

```diff
@@ -89,9 +90,11 @@ var AppListItemComponent = React.createClass({
 
     let labelsWidth = 0;
     let numberOfVisibleLabels = 0;
+    let labelNodes = React.findDOMNode(refs.labels).querySelectorAll(".label");
 
-    refs.labels.props.children[0].find((label) => {
-      labelsWidth += DOMUtil.getOuterWidth(React.findDOMNode(refs[label.ref]));
+    // labelNodes is not an Array, but a NodeList
+    [].forEach.call(labelNodes, function (label) {
```

Comment:
I don't know, in this specific use case I don't see the mentioned "problems" with this approach as such... If you feel strongly against this "hack", we can get back to it later maybe?